### PR TITLE
openwrt: fix cross platform compiling.

### PR DESCRIPTION
**Pull request by PolynomialDivision**
Fix cross platform compiling.

### CODE REVIEW

File: lib/util/stats.c [hunk 1]
```diff
@@ -107,7 +108,7 @@ int stats_print(struct stats_record *stats_rec, struct stats_record *stats_prev)
 			return 0;
 
 		if (first) {
-			printf("Period of %fs ending at %lu.%06lu\n", period,
+			printf("Period of %fs ending at %" PRId64 ".%06lu\n", period,
```
Comment (tohojo) [inline lib/util/stats.c:111]: Let's be consistent and replace the second `lu` here as well...

**Review by tohojo — CHANGES_REQUESTED** [review]
Thank you for the fixes! Only one nit...

**Comment by PolynomialDivision** [comment]
@tohojo Done. Also switched to PRIu64 for unsigned.

File: lib/util/stats.c (hunk 1 above)
Comment (PolynomialDivision) [reply lib/util/stats.c]: Does not work:
```
stats.c: In function 'stats_print':
stats.c:111:32: error: format '%llu' expects argument of type 'long long unsigned int', but argument 4 has type 'long int' [-Werror=format=]
  111 |                         printf("Period of %fs ending at %" PRIu64 ".%06" PRIu64 "\n", period,
      |                                ^~~~~~~~~~~~~~~~~~~~~~~~~~~
  112 |                                t.tv_sec, t.tv_nsec / 1000);
      |                                          ~~~~~~~~~~~~~~~~
      |                                                    |
      |                                                    long int
In file included from stats.c:9:
```

Comment (PolynomialDivision) [reply lib/util/stats.c]: I restructured now the PR:
https://github.com/xdp-project/xdp-tools/pull/206/commits/6ebbbb221b8edf19a4b6262a51fa83710a4e47c8

**Comment by PolynomialDivision** [comment]
It was said that using PRIu64 is wrong:
https://github.com/openwrt/openwrt/pull/10229#issuecomment-1179573186

**Comment by tohojo** [comment]
Right, so how about we just add casts to long and use `%ld` everywhere?

**Comment by PolynomialDivision** [comment]
> Right, so how about we just add casts to long and use `%ld` everywhere?

Updated the PR

**Review by tohojo — APPROVED** [review]

**Comment by tohojo** [comment]
Thank you for the fixes!

**Pull request merged**